adcc/AdcMatrix.py

```diff
@@ -112,19 +155,72 @@ def __init__(self, method, hf_or_mp, block_orders=None, intermediates=None):
         # Build the blocks and diagonals
         with self.timer.record("build"):
             variant = None
-            if method.is_core_valence_separated:
+            if self.is_core_valence_separated:
                 variant = "cvs"
             self.blocks_ph = {  # TODO Rename to self.block in 0.16.0
                 block: ppmatrix.block(self.ground_state, block.split("_"),
                                       order=order, intermediates=self.intermediates,
                                       variant=variant)
-                for block, order in block_orders.items() if order is not None
+                for block, order in self.block_orders.items() if order is not None
             }
             self.__diagonal = sum(bl.diagonal for bl in self.blocks_ph.values()
                                   if bl.diagonal)
             self.__diagonal.evaluate()
             self.__init_space_data(self.__diagonal)
 
+    def __iadd__(self, other):
+        """In-place addition of an :py:class:`AdcExtraTerm`
+
+        Parameters
+        ----------
+        other : AdcExtraTerm
+            the extra term to be added
+        """
+        if isinstance(other, list):
+            for k in other:
+                self += k
+            return self
+        if not isinstance(other, AdcExtraTerm):
+            raise TypeError("Can only add AdcExtra term or"
+                            " lists thereof to AdcMatrix.")
+        if not all(k in self.blocks_ph for k in other.blocks):
+            raise ValueError("Can only add to blocks of"
+                             " AdcMatrix that already exist.")
+        for sp in other.blocks:
+            ob = other.blocks[sp]
+            self.blocks_ph[sp].add_block(ob)
+        diag = sum(bl.diagonal for bl in other.blocks.values()
+                   if bl.diagonal)
+        # iadd does not work with numbers
+        self.__diagonal = self.__diagonal + diag
+        self.__diagonal.evaluate()
+        self.extra_terms.append(other)
```

Comment:
This feels to complicated and indirect. See my comment in the AdcBlock class below.

Reply:
Changed as per comment above.